### PR TITLE
Simplify Dockerfile and add a docker-compose file to create a container.

### PR DESCRIPTION
**Pull request by rbeucher**
Hi Luke,

I am using your docker file to run some tests. I had one but I tried to make it work with petsc 3.6 and 3.7 and it caused all sorts of issues...

I saw your makefile. One common way to do that is to use docker-compose.
I have attached a yml file that describe the port mapping, working directory, volumes etc...although I have not mount the /etc/password etc.

`docker-composer run --rm underworld` will start a container.

I don't know what your plans are with UW but it could be good to transfer your UW Earthbyte repo under the underworldcode organisation. That way we could keep track and maintain what was used for publications. Let me know what you think.

That could also be a good way to acknowledge all the work you have done.

Romain

### CODE REVIEW

File: docker-base/Dockerfile [hunk 1]
```diff
@@ -29,47 +28,38 @@ RUN apt-get update -qq \
...
-    && ./configure.py \
-        --hdf5-lib-dir=/usr/lib/x86_64-linux-gnu/hdf5/serial \
+ARG CONFIG_OPTIONS="--hdf5-lib-dir=/usr/lib/x86_64-linux-gnu/hdf5/serial \
         --hdf5-inc-dir=/usr/include/hdf5/serial \
         --petsc-dir=/petsc-3.5.4 \
         --with-debugging=0 \
-        --cflags="-Wno-error=format-security" \
+        --cflags='-Wno-error=format-security'"
+
+RUN git clone --depth=1 https://github.com/rbeucher/EarthByte_Underworld.git  \
```
Comment (LukeMondy) [inline docker-base/Dockerfile]: I would prefer that this git link points to whatever repo the Docker file is in - maybe we can have the URL as an ARG with the default as this repo (or if it moves, whatever repo that is).

**Review by LukeMondy — CHANGES_REQUESTED** [review]

**Comment by LukeMondy** [comment]
Hi Romain,

Yes, happy to have the repo forked over to the UnderworldCode organisation, sounds like a good idea!

I am aware of Docker-compose, but preferred the Makefile, since (at the time - not sure about now), it doesn't support the `-u "$(shell id -u)"` directive (along with the `/etc/passwd`) - which makes Underworld run under your host's local username, rather than root in the container.

Does docker-compose write out the outputs as root?

We could also make a user in the container (similar to Underworld2), and then (so long as the UIDs match up), it also writes out non-root files.

File: docker-base/Dockerfile (hunk 1 above)
Comment (rbeucher) [reply docker-base/Dockerfile]: Oh I did not see it was pointing to my fork...

Comment (rbeucher) [reply docker-base/Dockerfile]: I have added an `UW_EARTHBYTE` ARG. What do you think?

Comment (LukeMondy) [reply docker-base/Dockerfile]: Maybe `UW_GIT_URL`? No need to get more Earthbyte in there than necessary.

**Comment by rbeucher** [comment]
I believe you can do that with docker-compose version 3.0. You can just pass a user.
https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir

**Comment by rbeucher** [comment]
There is a thread here:
https://forums.docker.com/t/how-can-we-add-uid-and-gid-in-stack-file-or-compose-file/68261

**Comment by LukeMondy** [comment]
Can you do a test with the docker-compose user stuff?

**Comment by rbeucher** [comment]
passing `user: "uid"` works fine... the username does not... I think it looks if a user with the same name exist in the container.

**Comment by LukeMondy** [comment]
Yeah, you may need to mount the `/etc/passwd` file, so it can see the username/uid.

**Comment by rbeucher** [comment]
No I tried that.

**Comment by LukeMondy** [comment]
Well, I can't see much benefit of moving to Docker-compose if that doesn't work, given that not being able to delete files (because they're owned by `root`) is a big stumbling block for new users.

**Comment by rbeucher** [comment]
Well the filesystem works with numeric IDs so I think it makes sense to use that. We can rkbabky pass that as ${id -u username) but I haven't tried. I will have a look later on. The think is you won't get the name in the docker container if it has not been set. So maybe the makefile is better for that. I am a bit uncomfortable with the idea of mounting /etc/profile...

**Comment by LukeMondy** [comment]
I don't really mind what shows up within the container, just that the files have the right permissions.

Mounting `/etc/passwd`, as read-only, is safe :)

**Comment by LukeMondy** [comment]
I'm happy to merge in this change, without the Makefile deletion/docker-compose addition.
We could put that in a different PR and discuss it further there?

**Comment by LukeMondy** [comment]
The new commits seem to undo the changes to the Dockerfile that we want (e.g., `UW_GIT_URL`)?

Could you also squash the commits down into 1 or 2 commits?

Thanks!

**Comment by rbeucher** [comment]
The `UW_GIT_URL` is there. I have squashed the commits.

**Comment by LukeMondy** [comment]
Oh I see, it's in a new folder!
I don't see any reason to duplicate the dockerfile though?

**Comment by rbeucher** [comment]
No you are right. What about this?

**Comment by LukeMondy** [comment]
Cool, looks good.
If you could squash the commits down again to 1 commit, that'd be great!

**Comment by rbeucher** [comment]
You did test my git skills here...hhahah! That was fun!

**Pull request merged**